lib/src/facts/linux/operating_system_resolver.cc

```diff
@@ -21,82 +22,186 @@ namespace bs = boost::system;
 
 namespace facter { namespace facts { namespace linux {
 
-    operating_system_resolver::data operating_system_resolver::collect_data(collection& facts)
+    // Return contents of the os-release file
+    // http://www.freedesktop.org/software/systemd/man/os-release.html
+    static map<string, string> key_value_file(string file)
```

Comment:
Possible, but I'd have to read the file twice: once to determine check if it's Cumulus, CoreOS, NXOS, or another Linux variant, and again to read the relevant data for that OS.

However, most variants only read it once, and twice isn't too bad.